From: Jan Kara <jack@suse.cz>
Subject: blk: Fix kabi due to blk_trace_mutex addition
References: bsc#1159285
Patch-mainline: Never, kabi fix

struct request_queue is always dynamically allocated. Move new field to the
end.

Signed-off-by: Jan Kara <jack@suse.cz>

---
 include/linux/blkdev.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@ -572,7 +572,6 @@ struct request_queue {
 	int			node;
 #ifdef CONFIG_BLK_DEV_IO_TRACE
 	struct blk_trace __rcu	*blk_trace;
-	struct mutex		blk_trace_mutex;
 #endif
 	/*
 	 * for flush operations
@@ -622,6 +621,9 @@ struct request_queue {
 	u64			write_hints[BLK_MAX_WRITE_HINTS];
 #ifndef __GENKSYMS__
 	struct mutex		mq_freeze_lock;
+#ifdef CONFIG_BLK_DEV_IO_TRACE
+	struct mutex		blk_trace_mutex;
+#endif
 #endif
 };
 
